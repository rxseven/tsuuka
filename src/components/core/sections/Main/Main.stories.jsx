import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Main from './index';

const { text } = mock;
const title = 'Components/Core/Sections/Main';
const component = <Main>{text}</Main>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
