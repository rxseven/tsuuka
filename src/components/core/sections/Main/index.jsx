import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

import Grid from 'components/common/base/Grid';

const Frame = styled.main`
  margin-top: 3.875rem;
`;

const propTypes = exact({
  children: PropTypes.element.isRequired
});

function Main({ children }) {
  return (
    <Frame>
      <Grid.Container>
        <Grid.Row>
          <Grid.Column>{children}</Grid.Column>
        </Grid.Row>
      </Grid.Container>
    </Frame>
  );
}

Main.prototype = propTypes;

export default Main;
