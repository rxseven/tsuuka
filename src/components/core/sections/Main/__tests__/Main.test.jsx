import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Main from '../index';

describe('<Main />', () => {
  // Arrange
  const props = { children: 'content' };
  const component = <Main {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
