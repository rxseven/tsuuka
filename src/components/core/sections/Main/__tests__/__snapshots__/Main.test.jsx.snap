// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Snapshot tests should render correctly 1`] = `
<main
  className="sc-bdVaJa gFhbTF"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        content
      </div>
    </div>
  </div>
</main>
`;
